bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -114,38 +123,38 @@ if args.make_signal_kde:
                          (len(mass2), N_original, str(args.min_mass)))
     else:
         idx = numpy.full(N_original, True)
-    mass2_s = mass2
-    mass1_s = data_signal['mass1'][idx]
-    assert min(mass1_s - mass2_s) > 0
-    eta_s = data_signal['eta'][idx]
-    chi_eff_s  = data_signal['chi_eff'][idx]
-    mchirp_s = data_signal['mchirp'][idx]
+    mass1 = data_signal['mass1'][idx]
+    assert min(mass1 - mass2) > 0
+    eta = data_signal['eta'][idx]
+    chi_eff  = data_signal['chi_eff'][idx]
+    mchirp = data_signal['mchirp'][idx]
 
-    param0 = eval(args.fit_param[0][:])
-    param1 = eval(args.fit_param[1][:])
-    param2 = eval(args.fit_param[2][:])
-    if args.log_param == ['False', 'False', 'False']:
-        sample_s = numpy.vstack((param0, param1, param2)).T
-    elif args.log_param == ['True', 'False', 'False']:
-        sample_s = numpy.vstack((numpy.log(param0), param1, param2)).T
-    elif args.log_param == ['True', 'True', 'False']:
-        sample_s = numpy.vstack((numpy.log(param0), numpy.log(param1), param2)).T
-    else:
-        raise ValueError("invalid log_param argument, use 'True', or 'False'")
+    for param, slog in zip(args.fit_param, args.log_param):
+        data = data_signal[param]
+        if slog in ['false', 'False', 'FALSE']:
+            logging.info('Using param: %s', param)
+            paramvals.append(data)
+        elif slog in ['true', 'True', 'TRUE']:
+            logging.info('Using log param: %s', param)
+            paramvals.append(numpy.log(data))
+        else:
+            raise ValueError("invalid log param argument, use 'true', or 'false'")
 
-    grid_pts_s = numpy.array((numpy.log(mchirp_n), eta_n, chi_eff_n)).T
+    sample2 = numpy.vstack((paramvals)).T
     logging.info('Starting optimization of signal KDE parameters')
-    optbw_s, optalpha_s = optimizedparam(sample_s, nfold=2)
+    optbw, optalpha = optimizedparam(sample2, nfold=eval(args.nfold_signal))
     logging.info('Evaluating signal KDE')
-    kde_data_s = kde_awkde(sample_s, grid_pts_s, alp=optalpha_s, gl_bandwidth=optbw_s)
+    kde_data = kde_awkde(sample2, grid_pts, alp=optalpha, gl_bandwidth=optbw)
  
     hdf = h5py.File(args.output_file, 'w')
-    hdf.create_dataset('kde_sig', data=kde_data_s)
-    hdf.attrs['signal_kde'] = 'kde_sig'
-    hdf.create_dataset('mass1', data=mass1_s)
-    hdf.create_dataset('mass2', data=mass2_s)
-    hdf.create_dataset('mchirp', data=mchirp_s)
-    hdf.create_dataset('eta', data=eta_s)
-    hdf.create_dataset('chi_eff', data=chi_eff_s)
-    hdf.create_dataset('grid_pts', data=grid_pts_s)
+    hdf.create_dataset('kde_sig', data=kde_data)
+    hdf.attrs['kde_signal'] = 'data_signal'
+    hdf.attrs['fit_param'] = 'args.fit_param'
+    hdf.attrs['log_param'] = 'args.log_param'
+    hdf.create_dataset('mass1', data=mass1)
```

Comment:
We want the output file to have the same set of datasets in both the signal and template KDE cases, ie the datasets mass1, mass2, spin1z, spin2z, approximant, f_lower etc (whatever was in the original bank file) should be present in the output file. As suggested in another comment the way to do this without repeating lines of code is to open the output file earlier on and immediately write the original template file datasets into it.